jma_customizations.module

```diff
@@ -552,3 +555,71 @@ function jma_customizations_views_data_alter(&$data) {
     }
   }
 }
+
+function solr_address_geocoder($form, FormStateInterface $form_state) {
+  $address = [];
+  $values = $form_state->getValues();
+  foreach (['street_address', 'city', 'postal_code'] as $key) {
+    if (!empty($values[$key])) {
+      $address[] = $values[$key];
+    }
+  }
+  $address = implode(',', $address);
+  if (!empty($address)) {
+    $result = geocode($address);
+    if (!$result || $result['status'] != 'OK') {
+      $form_state->setErrorByName('street_address', t('The address for proximity search was not found'));
+    }
+    elseif (empty($result['results'][0]['geometry']['location'])) {
+      $form_state->setValue('center_lat', $result['results'][0]['geometry']['location']['lat']);
+      $form_state->setValue('center_long', $result['results'][0]['geometry']['location']['long']);
+    }
+  }
+}
+
+function geocode($address) {
+  $config = \Drupal::config('geolocation_google_maps.settings');
+  if (empty($address)) {
+    return FALSE;
+  }
+
+  if (!empty($config->get('google_maps_base_url'))) {
+    $request_url = $config->get('google_maps_base_url');
+  }
+  else {
+    $request_url = GoogleMaps::$GOOGLEMAPSAPIURLBASE;
+  }
+  $request_url .= '/maps/api/geocode/json?address=' . urlencode($address);
+
+  if (!empty($config->get('google_map_api_server_key'))) {
+    $request_url .= '&key=' . $config->get('google_map_api_server_key');
+  }
+  elseif (!empty($config->get('google_map_api_key'))) {
```

Comment:
@monishdeb isn't this just redundant / silly

Reply:
umm there is a difference, as it's fetching the key from google_map_api_server_key or google_map_api_key setting. In our setup, later is used, but always good to have flexible code isn't it!!

Reply:
oh i didn't spot the word server, why is there a diffence? module versions?